### PR TITLE
Prevent sway from duplicating on a failed fork

### PR DESCRIPTION
**Pull request by 4e554c4c**
Also remove a useless `sway_log` and replace it with a pipe

### CODE REVIEW

**Comment by ascent12** [comment]
You should probably be using _Exit() instead of exit() when exiting from a child.

File: sway/config.c [hunk 1]
```diff
@@ -922,14 +930,17 @@ static void invoke_swaybar(struct bar_config *bar) {
...
 			execvp(cmd[0], cmd);
+			exit(EXIT_SUCCESS);
 		} else {
 			// run custom swaybar
 			int len = strlen(bar->swaybar_command) + strlen(bar->id) + 5;
 			char *command = malloc(len * sizeof(char));
 			if (!command) {
-				sway_log(L_ERROR, "Unable to allocate swaybar command string");
-				return;
+				write(filedes[1], "Unable to allocate swaybar command string", 42);
```
Comment (SirCmpwn) [inline sway/config.c]: Use a constant and sizeof

File: sway/config.c [hunk 2]
```diff
@@ -940,10 +951,22 @@ static void invoke_swaybar(struct bar_config *bar) {
...
 			};
 
+			close(filedes[1]);
 			execvp(cmd[0], cmd);
 			free(command);
+			exit(EXIT_SUCCESS);
+		}
+	}
+	close(filedes[0]);
+	char *buf = malloc(1024);
```
Comment (SirCmpwn) [inline sway/config.c]: Can you write an int with the length, then allocate an appropriately sized buffer for it?

**Comment by 4e554c4c** [comment]
@ascent12 I changed it. What's the difference?

File: sway/config.c [hunk 3]
```diff
@@ -940,8 +954,22 @@ static void invoke_swaybar(struct bar_config *bar) {
...
+			close(filedes[1]);
 			execvp(cmd[0], cmd);
 			free(command);
+			_exit(EXIT_SUCCESS);
+		}
+	}
+	close(filedes[0]);
+	int len;
+	if(read(filedes[1], &len, sizeof(int)) == sizeof(int)) {
+		char *buf = malloc(len);
```
Comment (SirCmpwn) [inline sway/config.c]: You need to free buf when you're done with it

Comment (4e554c4c) [inline sway/config.c]: Thanks for catching that.

**Pull request merged**